ngraph/core/include/ngraph/validation_util.hpp

```diff
@@ -226,6 +226,31 @@ namespace ngraph
                                    std::map<RawNodeOutput, HostTensorPtr>& output_tensor_map,
                                    const OutputVector& outputs);
 
+    bool could_propagate(const Output<Node>& output, NodeVector& order);
```

Comment:
All functions declarations in header files are considered as public and must follow backward compatibility rules. So can we reduce the number of exposed function? I think at least `could_propagate` should be moved in cpp only.